src/main/java/org/myrobotlab/math/MapperLinear.java

```diff
@@ -53,11 +53,12 @@ final public Double calcOutput(Double in) {
         return 0.0;
       }
 
-      if (!inverted) {
-        c = minY + ((in - minX) * (maxY - minY)) / (maxX - minX);
-      } else {
-        c = minY + ((in - maxX) * (maxY - minY)) / (minX - maxX);
-      }
+      c = minY + ((in - minX) * (maxY - minY)) / (maxX - minX);
+      
+      if (inverted) {
+        c = c * -1;

```

Comment:
I don't think inverted means multiply by -1.  if output limit is from 0 to 5.   and we invert the control.  let's say "c" = 2.0 ... if we invert, I think the expected behavior should be 3.0  .. not -2.0

Reply:
Think about the normal case to be   "the output is y above the minY"
Then the inverted case is  "the output is y below the maxY"  ...

Reply:
Thanks, excellent catch - the devil made me do that "optimization"
I'll put in an "asymmetrical"  invert junit test.
Keep the tomatoes coming !
![local_08_temp-1346310620-503f11dc-620x348](https://user-images.githubusercontent.com/1266144/44418819-4042d300-a52e-11e8-9fe8-906c871031a7.jpg)